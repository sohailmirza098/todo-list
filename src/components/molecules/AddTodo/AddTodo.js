import React, { Component } from 'react'
import PropTypes from 'prop-types';
import TextInput from '../../atoms/TextInput/TextInput';
import ButtonInput from '../../atoms/ButtonInput/ButtonInput';

export class AddTodo extends Component {
    state = {
        title: ''
    }

    onChange = (e) => {
        this.setState({
            [e.target.name]: e.target.value
        })
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.addTodo(this.state.title);
        this.setState({ title: ''})
    }

    render() {
        return (
            <form style={{display:'flex'}} onSubmit={this.onSubmit}>
                <TextInput title={this.state.title} onChange={this.onChange} />
                <ButtonInput />
            </form>
        )
    }
}

// PropTypes
AddTodo.propTyes = {
    addTodo: PropTypes.func.isRequired
}

export default AddTodo
