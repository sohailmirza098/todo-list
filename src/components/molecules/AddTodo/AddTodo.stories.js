import React from 'react';
import AddTodo from "./AddTodo";
import { action } from '@storybook/addon-actions';
// import { text } from '@storybook/addon-knobs/react';

export default {
  title: 'AddTodo',
  component: AddTodo,
};

export const input = () => {
    return(<AddTodo />);
};
