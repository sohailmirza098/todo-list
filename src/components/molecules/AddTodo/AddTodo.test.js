import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddTodo from "./AddTodo";

describe("AddTodo testing", () => {

  test("Matches the snapshot", () => {
    const input = create(<AddTodo />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("AddTodo is present", () => {
    const {getByTestId}=  render(<AddTodo />);
    const addTodo = getByTestId("addtodoid");
    expect(addTodo).toBeInTheDocument();
  });

//   test("Header text content", () => {
//     const {getByTestId}=  render(<Header />);
//     const header = getByTestId("headerid");
//     expect(header.textContent).toEqual("TodoList")
//   });

});