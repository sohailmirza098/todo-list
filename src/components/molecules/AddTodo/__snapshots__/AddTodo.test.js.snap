exports[`AddTodo testing Matches the snapshot 1`] = `
<form
  data-testid="addtodoid"
  onSubmit={[Function]}
  style={
    Object {
      "display": "flex",
    }
  }>
  <input
    autoComplete="off"
    data-testid="texttodo"
    name="title"
    onChange={[Function]}
    placeholder="Add Todo..."
    style={
      Object {
        "flex": "10",
        "padding": "5px",
      }
    }
    type="text"
    value="" />
</form>
`;
